CMakeLists.txt

```diff
@@ -254,6 +254,17 @@ if(UMF_BUILD_EXAMPLES)
     add_subdirectory(examples)
 endif()
 
+# Conditional configuration for Level Zero provider
+if(UMF_BUILD_LEVEL_ZERO_PROVIDER)
+    set(OPTIONAL_SYMBOLS "umfLevelZeroMemoryProviderOps")
+else()
+    set(OPTIONAL_SYMBOLS "")
+endif()
+
+# Configure the DEF file based on whether Level Zero provider is built
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/src/libumf.def"
+               "${CMAKE_CURRENT_BINARY_DIR}/src/libumf.def" @ONLY)
```

Comment:
hmm, is it working? I believe we're using .map/.def from source_dir, not from bin_dir (pls look how `WINDOWS_DEF_FILE` is used)